src/pykeen/datasets/cli.py

```diff
@@ -205,5 +215,77 @@ def verify(dataset: str):
     click.echo(df.to_markdown())
 
 
+@main.command()
+@verbose_option
+@click.option("-d", "--dataset", help="Regex for filtering datasets by name")
+@click.option("-m", "--max-triples", type=int, default=None)
+def expected_metrics(dataset: str, max_triples: Optional[int]):
+    """Compute expected metrics for all datasets (matching the given pattern)."""
+    directory = PYKEEN_DATASETS
+    df_data: List[Tuple[str, str, str, str, float]] = []
+    for _dataset_name, dataset_cls in _iter_datasets(regex_name_filter=dataset, max_triples=max_triples):
+        if dataset_cls is OGBWikiKG:
+            click.echo("Skip OGB WikiKG")
+            continue
+        dataset_instance = get_dataset(dataset=dataset_cls)
+        dataset_name = dataset_instance.__class__.__name__.lower()
+        d = directory.joinpath(dataset_name, "analysis")
+        d.mkdir(parents=True, exist_ok=True)
+        expected_metrics = dict()
+        for key, factory in dataset_instance.factory_dict.items():
+            if key == "training":
+                additional_filter_triples = None
+            elif key == "validation":
+                additional_filter_triples = dataset_instance.training.mapped_triples
+            elif key == "testing":
+                additional_filter_triples = [
+                    dataset_instance.training.mapped_triples,
+                ]
+                if dataset_instance.validation is None:
+                    click.echo(f"WARNING: {dataset_name} does not have validation triples!")
+                else:
+                    additional_filter_triples.append(dataset_instance.validation.mapped_triples)
+            else:
+                raise AssertionError(key)
+            df = get_candidate_set_size(
+                mapped_triples=factory.mapped_triples,
+                additional_filter_triples=additional_filter_triples,
+            )
+            output_path = d.joinpath(f"{key}_candidates.tsv.gz")
+            df.to_csv(output_path, sep="\t", index=False)
+
+            # expected metrics
+            this_metrics = dict()
+            for label, sides in dict(
+                head=["head"],
+                tail=["tail"],
+                both=["head", "tail"],
+            ).items():
+                candidate_set_sizes = df[[f"{side}_candidates" for side in sides]]
+                this_metrics[label] = {
+                    "mean_rank": expected_mean_rank(candidate_set_sizes),
+                    **{f"hits_at_{k}": expected_hits_at_k(candidate_set_sizes, k=k) for k in (1, 3, 5, 10)},
```

Comment:
add larger ones like 100 and 1000 if dataset is big enough

Reply:
8d66ef8